src/awesome/client.rs

```diff
@@ -42,9 +43,60 @@ impl<'lua> ToLua<'lua> for Client<'lua> {
     }
 }
 
-impl UserData for ClientState {}
+impl UserData for ClientState {
+    fn add_methods(methods: &mut UserDataMethods<Self>) {
+        methods.add_method("geometry", |lua, state, (geometry,): (Value,)| {
+            match geometry {
+                Value::Table(table) => {
+                    let area = Area {
+                        origin: Origin { x: table.get("x")?, y: table.get("y")? },
+                        size: Size { width: table.get("width")?, height: table.get("height")? }
+                    };
+
+                    state.view.move_resize(area);
+
+                    Ok(Value::Nil.to_lua(lua))
+                },
+                _ => {
+                    let table = lua.create_table()?;
+                    let geometry = state.view.geometry();
+                    let Origin { x, y } = *state.view.origin.lock().unwrap();
+                    table.set("x", geometry.origin.x + x)?;
+                    table.set("y", geometry.origin.y + y)?;
+                    table.set("width", geometry.size.width)?;
+                    table.set("height", geometry.size.height)?;
+                    Ok(table.to_lua(lua))
+                }
+            }
+        });
+
+        methods.add_meta_function(MetaMethod::Index, |lua, (class, index): (ClientState, Value)| {
+            match index {
+                Value::String(value) => {
+                    match value.to_str() {
+                        Ok("title") => {
+                            Ok(class.view.title().to_lua(lua))
+                        },
+                        _ => Ok(Value::Nil.to_lua(lua))
+                    }
+                },
+                _ => Ok(Value::Nil.to_lua(lua))
+            }
+        });
+    }
+}
+
+pub fn init<'lua>(lua: &'lua Lua, server: &Server) -> rlua::Result<Class<'lua>> {
+    let clients: &mut Vec<Client> = &mut Vec::default();
+
+    for view in &server.views {
+        let mut client = Client::cast(Client::new(lua)?)?;
+        client.init_client(view.clone())?;
+        clients.push(client);
+    }
+
+    lua.set_named_registry_value(CLIENTS_HANDLE, clients.clone().to_lua(lua)?)?;
```

Comment:
Why are you cloning the vector here? It looks like you don't use it anywhere else?